src/EntityManagerFactory.php

```diff
@@ -318,14 +325,32 @@ protected function setCustomFunctions(Configuration $configuration)
      */
     protected function setCacheSettings(Configuration $configuration)
     {
-        if ($namespace = $this->config->get('doctrine.cache.namespace', null)) {
-            $this->cache->driver()->setNamespace($namespace);
-        }
+        $configuration->setQueryCacheImpl($this->applyNamedCacheConfiguration('query'));
+        $configuration->setResultCacheImpl($this->applyNamedCacheConfiguration('result'));
+        $configuration->setMetadataCacheImpl($this->applyNamedCacheConfiguration('metadata'));
 
         $this->setSecondLevelCaching($configuration);
     }
 
     /**
+     * @param  string $cacheName
+     * @return mixed
+     */
+    private function applyNamedCacheConfiguration($cacheName)
+    {
+        $defaultDriver    = $this->config->get('doctrine.cache.default', $this->defaultCache['type']);
```

Comment:
Replace `$this->defaultCache['type']` by `'array'`
Remove `$this->defaultCache['namespace']`